### PR TITLE
Feature4

### PR DESCRIPTION
**Pull request by mudasiri**
In this Pull request I have done the features base on requirements of project below:

### Project requirements
- Create a table named `vets` with the following columns:
  - `id`: integer (set it as autoincremented PRIMARY KEY)
  - `name`: string
  - `age`: integer
  - `date_of_graduation`: date
- There is a **many-to-many** relationship between the tables `species` and `vets`: a vet can specialize in multiple species, and a species can have multiple vets specialized in it. Create a "join table" called `specializations` to handle this relationship.
- There is a **many-to-many** relationship between the tables `animals` and `vets`: an animal can visit multiple vets and one vet can be visited by multiple animals. Create a "join table" called `visits` to handle this relationship, it should also keep track of the date of the visit.
- Insert the following data for vets:
  - Vet William Tatcher is 45 years old and graduated Apr 23rd, 2000.
  - Vet Maisy Smith is 26 years old and graduated Jan 17th, 2019.
  - Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
  - Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.
- Insert the following data for specialties:
  - Vet William Tatcher is specialized in Pokemon.
  - Vet Stephanie Mendez is specialized in Digimon and Pokemon.
  - Vet Jack Harkness is specialized in Digimon.
- Insert the following data for visits:
  - Agumon visited William Tatcher on May 24th, 2020.
  - Agumon visited Stephanie Mendez on Jul 22th, 2020.
  - Gabumon visited Jack Harkness on Feb 2nd, 2021.
  - Pikachu visited Maisy Smith on Jan 5th, 2020.
  - Pikachu visited Maisy Smith on Mar 8th, 2020.
  - Pikachu visited Maisy Smith on May 14th, 2020.
  - Devimon visited Stephanie Mendez on May 4th, 2021.
  - Charmander visited Jack Harkness on Feb 24th, 2021.
  - Plantmon visited Maisy Smith on Dec 21st, 2019.
  - Plantmon visited William Tatcher on Aug 10th, 2020.
  - Plantmon visited Maisy Smith on Apr 7th, 2021.
  - Squirtle visited Stephanie Mendez on Sep 29th, 2019.
  - Angemon visited Jack Harkness on Oct 3rd, 2020.
  - Angemon visited Jack Harkness on Nov 4th, 2020.
  - Boarmon visited Maisy Smith on Jan 24th, 2019.
  - Boarmon visited Maisy Smith on May 15th, 2019.
  - Boarmon visited Maisy Smith on Feb 27th, 2020.
  - Boarmon visited Maisy Smith on Aug 3rd, 2020.
  - Blossom visited Stephanie Mendez on May 24th, 2020.
  - Blossom visited William Tatcher on Jan 11th, 2021.
- Write queries to answer the following:
  - [Who was the last animal seen by William Tatcher?](https://github.com/mudasiri/Vet-Clinic/blob/Feature4/images/day%204/seen%20by%20william%20tacher%20new.png)
  - [How many different animals did Stephanie Mendez see?](https://github.com/mudasiri/Vet-Clinic/blob/Feature4/images/day%204/stephenie%20mendez%20new.png)
  - [List all vets and their specialties, including vets with no specialties.](https://github.com/mudasiri/Vet-Clinic/blob/Feature4/images/day%204/vets%20no%20specialities.png)
  - List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
  - [What animal has the most visits to vets?](https://github.com/mudasiri/Vet-Clinic/blob/Feature4/images/day%204/animal_mostvisits.png)
  - [Who was Maisy Smith's first visit?](https://github.com/mudasiri/Vet-Clinic/blob/Feature4/images/day%204/maisy%20first%20visit.png)
  - Details for most recent visit: animal information, vet information, and date of visit.
  - How many visits were with a vet that did not specialize in that animal's species?
  - What specialty should Maisy Smith consider getting? Look for the species she gets the most.
- **Take a screenshot of the results of your queries.**
- Remember to add the screenshot with the results of your queries to your Pull Request description.

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -36,4 +36,116 @@ ADD FOREIGN KEY (species_id) REFERENCES species(id);
...
+    date_of_graduation DATE
+);
+
+/*There is a many-to-many relationship between 
+the tables species and vets: a vet can specialize 
+in multiple species, and a species can have multiple 
+vets specialized in it. Create a "join table" called 
+specializations to handle this relationship.*/
+
+create table specializations (species_id INT, vets_id INT);
```
Comment (Davidosky007) [inline schema.sql:55]: * Kindly add all these data below 👇 as it is a[ requirement ](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_join_table.md#:~:text=Insert%20the%20following,Jan%2011th%2C%202021.)

> Insert the following data for vets:
> Vet William Tatcher is 45 years old and graduated Apr 23rd, 2000.
> Vet Maisy Smith is 26 years old and graduated Jan 17th, 2019.
> Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
> Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.
> Insert the following data for specialties:
> Vet William Tatcher is specialized in Pokemon.
> Vet Stephanie Mendez is specialized in Digimon and Pokemon.
> Vet Jack Harkness is specialized in Digimon.
> Insert the following data for visits:
> Agumon visited William Tatcher on May 24th, 2020.
> Agumon visited Stephanie Mendez on Jul 22th, 2020.
> Gabumon visited Jack Harkness on Feb 2nd, 2021.
> Pikachu visited Maisy Smith on Jan 5th, 2020.
> Pikachu visited Maisy Smith on Mar 8th, 2020.
> Pikachu visited Maisy Smith on May 14th, 2020.
> Devimon visited Stephanie Mendez on May 4th, 2021.
> Charmander visited Jack Harkness on Feb 24th, 2021.
> Plantmon visited Maisy Smith on Dec 21st, 2019.
> Plantmon visited William Tatcher on Aug 10th, 2020.
> Plantmon visited Maisy Smith on Apr 7th, 2021.
> Squirtle visited Stephanie Mendez on Sep 29th, 2019.
> Angemon visited Jack Harkness on Oct 3rd, 2020.
> Angemon visited Jack Harkness on Nov 4th, 2020.
> Boarmon visited Maisy Smith on Jan 24th, 2019.
> Boarmon visited Maisy Smith on May 15th, 2019.
> Boarmon visited Maisy Smith on Feb 27th, 2020.
> Boarmon visited Maisy Smith on Aug 3rd, 2020.
> Blossom visited Stephanie Mendez on May 24th, 2020.
> Blossom visited William Tatcher on Jan 11th, 2021.

File: schema.sql [hunk 2]
```diff
@@ -36,4 +36,116 @@ ADD FOREIGN KEY (species_id) REFERENCES species(id);
...
+FROM visits 
+JOIN vets ON visits.vets_id = vets.id 
+JOIN animals ON visits.animals_id = animals.id 
+WHERE vets.name = 'William Tatcher' 
+ORDER BY date_of_visit DESC 
+LIMIT 1;
+
+
+--How many different animals did Stephanie Mendez see?--
+SELECT COUNT(animals.name)
```
Comment (Davidosky007) [inline schema.sql]: * Kindly move all your queries into the `qeuries.sql` file as its best practice

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @mudasiri 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Well written PR 👍 
* Good looking readme file 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql [hunk 3]
```diff
@@ -51,4 +51,39 @@ SET owner_id = (
...
+INSERT INTO specializations (species_id, vets_id) VALUES (2, 3);
+INSERT INTO specializations (species_id, vets_id) VALUES (2, 3);
+INSERT INTO specializations (species_id, vets_id) VALUES (1, 4);
+
+/*3*/
+INSERT INTO visits (animals_id, vets_id, date_of_visit) VALUES (1, 1, '2020-05-24');
+INSERT INTO visits (animals_id, vets_id, date_of_visits) VALUES (1, 3, '2020-07-22');
+INSERT INTO visits (animals_id, vets_id, date_of_visits) VALUES (2, 4, '2021-02-02');
+INSERT INTO visits (animals_id, vets_id, date_of_visits) VALUES (3, 2, '2020-01-05');
+INSERT INTO visits (animals_id, vets_id, date_of_visit) VALUES (3, 2, '2020-05-08');
```
Comment (BenMukebo) [inline data.sql]: - [ ]Please pay more attention when inserting our data for visits, here are the project requirements 👇🏻 if you pay more attention to the fifth insert you can see it has 'mar' as the month which is the short form of March, not May. Please fix that 👍🏻
 
image

- Your code might look like this 👇🏻 

`INSERT INTO visits (animal_id, vet_id, vist_date) VALUES ((SELECT id FROM animal WHERE name = 'Pikachu'), (SELECT id FROM vets WHERE name = 'Maisy Smith'), '2020-03-08');`

File: queries.sql [hunk 4]
```diff
@@ -85,4 +85,88 @@ FROM owners
...
+FROM visits 
+JOIN vets ON visits.vets_id = vets.id 
+JOIN animals ON visits.animals_id = animals.id 
+WHERE vets.name = 'William Tatcher' 
+ORDER BY date_of_visit DESC 
+LIMIT 1;
+
+
+--How many different animals did Stephanie Mendez see?--
+SELECT COUNT(animals.name)
```
Comment (BenMukebo) [inline queries.sql:101]: - [ ] According to the project requirements. Add a screenshot of the query results to the pull request description. This also helps to ensure that you run the queries with success. Kindly make this 💯 👍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
Hi  @mudasiri, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight :clap: 
- Gitflow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️


 ## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly, let's add correct information to the  `Show your support` and `Acknowledgments` section of the README file, This should have correct information. ✍🏻 

image


- [ ]  I'd recommend that you delete the Note: 👇🏻 in your License section as these are instructions on how to complete the License section.


image


You have submitted a project, that Linters check are not running in the GitHub/action
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by franclobo — COMMENTED** [review]
Hi @mudasiri, 

terminator

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have professional documentation. All main headers from the [README template](https://github.com/microverseinc/readme-template) should be present in your repo. Please edit your README with the correct information of the current project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by franclobo — COMMENTED** [review]
Hi @mudasiri, 

roja

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have a professional README file. Please 🙏🏽 remove the FAQ section if you don't have questions and answers about the current project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Reem-lab — APPROVED** [review]
# Hi @mudasiri,👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🔥🔥💯💯


## Highlights 🔥🔥🔥🔥
- Followed Gitflow ✅
- Great working on the functionality 🔥🔥




## Optional suggestions



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — COMMENTED** [review]
Hi Muda, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- great job creating the queries.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**